File: src/p1.TwoSum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

import java.util.*;

class TwoSum {
    // @lc code=start
    class Solution {
        public int[] twoSum(int[] nums, int target) {
            HashMap<Integer, Integer> record = new HashMap<>();
            for (int i = 0; i < nums.length; ++i) {
                int num = nums[i];

                if (record.containsKey(num)) {
                    return new int[] { record.get(num), i };
                } else {
                    record.put(target - num, i);
                }
            }

            return new int[] {};
        }
    }
    // @lc code=end

    public static void main(String[] args) {
        var s = (new TwoSum()).new Solution();
        var result = s.twoSum(new int[] { 2, 7, 11, 15 }, 9);
        System.out.println(Arrays.toString(result));
    }

}

File: src/p1.twoSum.kt
/*
 * @lc app=leetcode id=1 lang=kotlin
 *
 * [1] Two Sum
 */
import java.util.Arrays

// @lc code=start
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val record = mutableMapOf<Int, Int>()

        for ((i, num) in nums.withIndex()) {
            when (val matched = record[num]) {
                is Int -> return intArrayOf(matched, i)
                else -> record[target - num] = i
            }
        }

        return IntArray(0)
    }
}
// @lc code=end

fun main() {
    val s = Solution()
    val result = s.twoSum(intArrayOf(2, 7, 11, 15), 9)
    println(Arrays.toString(result))
}
